src/messaging/ExchangeMgr.cpp

```diff
@@ -104,6 +104,12 @@ CHIP_ERROR ExchangeManager::Shutdown()
     mMessageCounterSyncMgr.Shutdown();
     mReliableMessageMgr.Shutdown();
 
+    for (auto & ec : mContextPool)
+    {
+        // ExchangeContext leaked
+        assert(ec.GetReferenceCount() == 0);
```

Comment:
We shouldn't really be using `assert`.  I'm not going to block on that, but please file a followup to use something here that does not have the bad codesize effects `assert` does.